tests/js/profile/wearable/widget/wearable/NumberPicker/NumberPicker.js

```diff
@@ -50,10 +50,185 @@
+			assert.ok(widget.options.disabled, "option disabled is correct");
+		});
+
+		QUnit.test("_configure", 2, function (assert) {
+			var widget = new NumberPicker(),
+				element = document.getElementById("number-picker");
+
+			helpers.stub(widget, "_getOptions", function (param) {
+				assert.ok(true, "_getOptions");
+				assert.equal(param, element, "_getOptions: element is given");
+			});
+
+			widget._configure(element);
+		});
+
+		QUnit.test("_toggle", 4, function (assert) {
+			var widget = new NumberPicker();
+
+			widget.options = {
+				disabled: false
+			};
+			widget._ui = {
+				container: document.createElement("div")
+			}
+			widget._toggle(true);
+
+			assert.ok(widget.options.disabled, "widget has been disabled");
+			assert.ok(widget._ui.container.classList.contains("ui-number-picker-disabled"), "widget have disabled class");
+
+			widget.options = {
+				disabled: true
+			};
+			widget._ui = {
+				container: document.createElement("div")
+			}
+			widget._ui.container.classList.add("ui-number-picker-disabled");
+
+			widget._toggle(false);
+
+			assert.ok(!widget.options.disabled, "widget is not disabled");
+			assert.ok(!widget._ui.container.classList.contains("ui-number-picker-disabled"), "widget doesn't have disabled class");
+		});
+
+		QUnit.test("_refresh", 7, function (assert) {
+			var widget = new NumberPicker(),
+				element = document.getElementById("number-picker");
+
+			helpers.stub(widget, "_getOptions", function () {
+				assert.ok(true, "_getOptions");
+			});
+			helpers.stub(widget, "_setValue", function (value) {
+				assert.ok(true, "_setValue");
+				assert.equal(value, element.getAttribute("value"), "element.value is correct");
+			});
+			helpers.stub(widget, "_toggleCircleIndicator", function (value) {
+				assert.ok(true, "_toggleCircleIndicator");
+				assert.ok(!value, "value is false");
+			});
+			helpers.stub(widget, "_toggle", function (value) {
+				assert.ok(true, "_toggle");
+				assert.ok(value, "value is true");
+			});
+
+			element.setAttribute("value", 10);
+			widget.element = element;
+			widget.options = {
+				disabled: true
+			};
+
+			widget._refresh();
+		});
+
+		QUnit.test("_setValue", 12, function (assert) {
+			var widget = new NumberPicker(),
+				element = document.getElementById("number-picker");
+
+
+			widget.element = element;
+			widget.options = {
+				min: 34,
+				max: 60
+			}
+
+			// case 1
+			helpers.stub(widget, "_updateValue", function (value) {
+				assert.ok(true, "_updateValue");
+				assert.equal(value, 40, "value is correct");
+			});
+
+			widget._setValue(40);
+			assert.equal(element.value, 40, "Value has been properly set");
+			assert.equal(element.getAttribute("value"), "40", "Value has been properly set");
+
+			// case LOOP if value < min the value = max
+			helpers.stub(widget, "_updateValue", function (value) {
+				assert.ok(true, "_updateValue");
+				assert.equal(value, 60, "value is correct");
+			});
+			widget._setValue(10);
```

Comment:
Is it correct value?

Reply:
This test results from implementation of method (https://github.com/Samsung/TAU/blob/4470e5655a6416ea189945f8b8a2c1dbf5c05a07/src/js/profile/wearable/widget/wearable/NumberPicker.js#L374):
```
				// loop widget value;
				if (value > options.max) {
					value = options.min;
				} else if (value < options.min) {
					value = options.max;
				}
```

Reply:
Oh I see, it looks like the expected value.